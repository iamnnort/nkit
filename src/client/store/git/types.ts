import { CommonStore } from '../../lib/store/types';

export enum ActionTypes {
  LOAD_RELEASES_REQUEST = '@@git/LOAD_RELEASES_REQUEST',
  LOAD_RELEASES_SUCCESS = '@@git/LOAD_RELEASES_SUCCESS',
  LOAD_RELEASES_FAILURE = '@@git/LOAD_RELEASES_FAILURE',
}

export interface Git extends CommonStore {
  releases: GitRelease[];
}

export interface GitRelease {
  id: number;
  name: string;
  tag_name: string;
  html_url: string;
}
