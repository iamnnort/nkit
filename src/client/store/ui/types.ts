export enum ActionTypes {
  TOGGLE_SIDEBAR = '@@ui/TOGGLE_SIDEBAR',
}

export interface UiInterface {
  isSidebarOpened: boolean;
}
