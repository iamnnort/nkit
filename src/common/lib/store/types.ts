export interface CommonStore {
  isInitialLoaded: boolean;
  isLoading: boolean;
  error: string;
}

export enum CommonError {
  FAILED_REQUEST = 'Loading was failed',
}
