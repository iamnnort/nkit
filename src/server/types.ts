import * as express from 'express';
import i18next from 'i18next';

export interface Request extends express.Request {
  i18n: i18next.i18n;
}
