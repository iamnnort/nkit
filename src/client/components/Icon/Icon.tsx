import * as React from 'react';

import UnknownSvg from './_svg/unknown.svg';

interface IconsInterface {
  [key: string]: { [key: string]: JSX.Element };
}

const icons: IconsInterface = {
  unknown: { default: <UnknownSvg /> },
};

interface Props {
  type: string;
  variant?: string;
}

const IconComponent: React.SFC<Props> = ({ type, variant = 'default' }) => icons[type][variant];

export default IconComponent;
