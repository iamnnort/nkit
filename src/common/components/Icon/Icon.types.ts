export interface Icons {
  [key: string]: { [key: string]: JSX.Element };
}

export interface ComponentProps {
  type: string;
  variant?: string;
}
