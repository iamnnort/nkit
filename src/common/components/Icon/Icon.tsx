import * as React from 'react';

import { Icons, ComponentProps } from './Icon.types';

import { ReactComponent as CheckMark} from './__svg__/CheckMark.svg';
import { ReactComponent as Tag }  from './__svg__/Tag.svg';

const icons: Icons = {
  checkMark: { default: <CheckMark /> },
  tag: { default: <Tag /> },
};

const IconComponent: React.FC<ComponentProps> = ({ type, variant = 'default' }) => icons[type][variant];

export default IconComponent;
