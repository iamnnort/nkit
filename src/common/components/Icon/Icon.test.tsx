import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import Icon from './Icon';
import { ComponentProps } from './Icon.types';

describe('IconComponent', () => {
  let wrapper: ReactWrapper;

  const props: ComponentProps = {
    type: 'checkMark',
    variant: 'default',
  };

  beforeEach(async () => {
    wrapper = mount(<Icon {...props} />);
  });

  it(`should not regret`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
