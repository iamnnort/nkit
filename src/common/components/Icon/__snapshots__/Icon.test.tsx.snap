// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent should not regret 1`] = `
<IconComponent
  type="checkMark"
  variant="default"
>
  <ForwardRef(SvgCheckMark)>
    <svg>
      CheckMark.svg
    </svg>
  </ForwardRef(SvgCheckMark)>
</IconComponent>
`;
